src/couch_replicator_httpc_pool.erl

```diff
@@ -70,23 +69,15 @@ handle_call(get_worker, From, State) ->
         callers = Callers,
         url = Url,
         limit = Limit,
-        busy = Busy,
-        free = Free
+        conns = Conns
     } = State,
-    case length(Busy) >= Limit of
+    case length(Conns) >= Limit of
     true ->
         {noreply, State#state{waiting = queue:in(From, Waiting)}};
     false ->
-        case Free of
-        [] ->
-           {ok, Worker} = ibrowse:spawn_link_worker_process(Url),
-           Free2 = Free;
-        [Worker | Free2] ->
-           ok
-        end,
+        {ok, Worker} = couch_replicator_connection:acquire(Url),
```

Comment:
Nit: Reply can be `error` as well? That's fine, we'd want to crash, but is it worth mentioning in a comment and/or log?